ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/validator/ModuleMemberValidator.java

```diff
@@ -51,25 +50,20 @@ public boolean evaluate(String source) {
         TextDocument document = TextDocuments.from(source);
         SyntaxTree tree = SyntaxTree.from(document);
         ModulePartNode node = tree.rootNode();
-        NodeList<ModuleMemberDeclarationNode> members = node.members();
-        if (members.size() > 1) {
-            return false;
+        // Ignore code segments including imports
+        if (node.imports().size() > 0) {
+            return true;
         }
 
-        boolean isIncomplete = NodeParser.parseModuleMemberDeclaration(source).apply(incompleteInputFinder);
-        if (!NodeParser.parseModuleMemberDeclaration(source).hasDiagnostics()) {
-            return false;
-        } else {
-            if (!source.endsWith(";") && !NodeParser.parseModuleMemberDeclaration(source + ";").hasDiagnostics()) {
-                return false;
-            }
+        Node parsedNode = NodeParser.parseModuleMemberDeclaration(source);
+        if (!parsedNode.hasDiagnostics()) {
+            return true;
         }
 
-        if (!isIncomplete) {
-            return  nextInValidator.evaluate(source);
-        } else {
+        if (!parsedNode.apply(incompleteInputFinder)) {
             return true;
         }
 
+        return nextInValidator.evaluate(source);
```

Comment:
We can merge these together

Reply:
Updated in 7d25e1dea188151fc8c5f84a1bf9156ce22ebc81